examples/vue-3-cli/src/stories/OverrideArgs.stories.js

```diff
@@ -0,0 +1,43 @@
+import OverrideArgs from './OverrideArgs.vue';
+
+// Emulate something that isn't serializable
+const icons = {
+  Primary: {
+    template: '<span>Primary Icon</span>',
+  },
+  Secondary: {
+    template: '<span>Secondary Icon</span>',
+  },
+};
+
+export default {
+  title: 'Example/Override Args',
+  component: OverrideArgs,
+  argTypes: {
+    // To show that other props are passed through
+    backgroundColor: { control: 'color' },
+    icon: {
+      control: {
+        type: 'select',
+        options: Object.keys(icons),
+      },
+      defaultValue: 'Primary',
+    },
+  },
+};
+
+const Template = (args, { argTypes }) => {
+  return {
+    props: Object.keys(argTypes),
+    components: { OverrideArgs },
+    template: '<override-args v-bind="$props" :icon="icon" />',
+    setup(props) {
+      return {
+        icon: icons[props.icon],
```

Comment:
Sorry for the slow feedback @phated. I think a computed property would be needed to do the calculation normally in Vue.  This example will only work if the component passing the icon is rendered from scratch every time, which when props change it would not normally.

e.g.
```js
setup(props) {
  const iconComponent = computed(() => icons[props.icon]);
  return { iconComputed };
}
```

See examples here
 https://codesandbox.io/s/passing-a-selected-component-7t1o3?file=/src/App.vue

Reply:
Hey @lee-chase, I don't actually write Vue, so I can't really evaluate your suggestion. During development of this feature, I wasn't able to use a component that was wrapped in a computed proxy without Vue 3 erroring out. Maybe that is different with the `<component :is="" />` syntax?

I wasn't able to make any changes to your codesandbox without babel failing for some reason, so I couldn't check there. Would you be able to send an example to the Vue 3 kitchen sink here? Or maybe just change my example to show a way it doesn't work?

Reply:
Sure, will have a look tomorrow.

Reply:
OK, @phated so I think the form of the story you had originally ought to have been good as 'args' is outside of the Template being tested and passed in as props. As we discovered this does not work, but I do not think that id down to Vue.

In fact, we could make the original work by using the following instead. In an ideal world, I'd be able to see how to fix this but I am new to the storybook code and short of time.

```js
  if (typeof args.icon === 'string') {
    args.icon = icons[args.icon];
  }
```

I have not found anything that feels particularly satisfying as they involve changing the template, however...

The simplest option that requires no changes elsewhere is to use the following setup(). While props.icon is read-only, context.props.icon is not. The advantage of this to me is that it does not confuse by returning an object that normally contains reactive values for the template.

```js
setup(props, context) {
  context.props.icon = icons[context.props.icon];
},
```

The option I suggested last time passes a reactive value back from setup. My only reason for preferring this to your solution is that it is code that is reactive without a full re-render. However, I dislike the fact icon is passed in via $props and directly.

```js
  template: '<override-args v-bind="$props" :icon="icon"/>',
 setup(props) {
    const iconOverride = computed(() => icons[props.icon]);
    return { icon: iconOverride };
  },
```

Which leads me to 
```js
  template: '<override-args v-bind="usingOverrides"/>',
  setup(props) {
    const usingOverrides = computed(() => ({ ...props }));
    usingOverrides.value.icon = icons[props.icon];
    return { usingOverrides };
  },
  ```

Strictly speaking, as we are not changing the value after setup we could go with ref instead of computed:
```js
  template: '<override-args v-bind="usingOverrides"/>',
  setup(props) {
    const usingOverrides = ref({ ...props });
    usingOverrides.value.icon = icons[props.icon];
    return { usingOverrides };
  },
  ```

It could even be abbreviated to the following:
```js
  template: '<override-args v-bind="usingOverrides"/>',
  setup(props) {
    return { usingOverrides: ref({ ...props, icon: icons[props.icon] }) };
  },
  ```

My favourite is the context change, but I'd really love a core Vue developer to tell me I wasn't being very naughty.
My second favourite is the last one.